### PR TITLE
Feature/31

### PR DESCRIPTION
**Pull request by AmrutaBorgaonkar**
This PR contains work for 2nd tab report i.e. Bridge Work Summary report, and changes to Bridge Data tab code in required areas. 1st tab sheet is used to compute 2nd tab report.(used data models created in 1st tab to compute and fill up 2nd tab.)

### CODE REVIEW

File: BridgeCareApp/BridgeCare/Services/SummaryReport/BridgeRateDeckAreaWorkSummary.cs [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+        /// <summary>
+        /// Fill work summary bridge poor on off rate and deck area sections
+        /// </summary>
+        /// <param name="worksheet"></param>
+        /// <param name="currentCell"></param>
+        /// <param name="simulationYears"></param>
+        /// <param name="simulationDataModels"></param>
+        public static void FillBridgeRateDeckAreaWorkSummarySections(ExcelWorksheet worksheet, CurrentCell currentCell, List<int> simulationYears, List<SimulationDataModel> simulationDataModels)
+        {
+            FillPoorBridgeOnOfRateSection(worksheet, currentCell, simulationYears, simulationDataModels);
```
Comment (ppator1) [inline BridgeCareApp/BridgeCare/Services/SummaryReport/BridgeRateDeckAreaWorkSummary.cs]: Should this be 'FillPoorBridgeOnOffRateSection'?

Comment (AmrutaBorgaonkar) [reply BridgeCareApp/BridgeCare/Services/SummaryReport/BridgeRateDeckAreaWorkSummary.cs]: Renamed, thanks.

File: BridgeCareApp/BridgeCare/Services/SummaryReport/BridgeRateDeckAreaWorkSummary.cs [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+using System.Linq;
+using System.Web;
+
+namespace BridgeCare.Services
+{
+    public static class BridgeRateDeckAreaWorkSummary
+    {
+        #region constants
+        const string BridgeCare = "Bridge Care";
+        const string Good = "Good";
```
Comment (ARA-Trans) [inline BridgeCareApp/BridgeCare/Services/SummaryReport/BridgeRateDeckAreaWorkSummary.cs]: This should be an enumeration or a set of classes. Preferably the latter.

File: BridgeCareApp/BridgeCare/Services/SummaryReport/BridgesCulvertsWorkSummary.cs [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+using System.Collections.Generic;
+using System.Drawing;
+using System.Linq;
+using System.Web;
+
+namespace BridgeCare.Services
+{
+    public static class BridgesCulvertsWorkSummary
+    {
+        #region constants
```
Comment (ARA-Trans) [inline BridgeCareApp/BridgeCare/Services/SummaryReport/BridgesCulvertsWorkSummary.cs]: Get rid of the region block.

File: BridgeCareApp/BridgeCare/Services/SummaryReport/CostBudgetsWorkSummary.cs [hunk 4]
```diff
@@ -0,0 +1,222 @@
...
+using OfficeOpenXml;
+using System;
+using System.Collections.Generic;
+using System.Drawing;
+using System.Linq;
+using System.Web;
+
+namespace BridgeCare.Services
+{
+    public static class CostBudgetsWorkSummary
```
Comment (ARA-Trans) [inline BridgeCareApp/BridgeCare/Services/SummaryReport/CostBudgetsWorkSummary.cs]: For all of these report classes, I don't think they can be static. They each contain state information that may change from user to user that is being written out to the report correct? What happens if multiple users try to run the same report at the same time?

File: BridgeCareApp/BridgeCare/Services/SummaryReport/BridgesCulvertsWorkSummary.cs (hunk 3 above)
Comment (AmrutaBorgaonkar) [reply BridgeCareApp/BridgeCare/Services/SummaryReport/BridgesCulvertsWorkSummary.cs]: Done.

File: BridgeCareApp/BridgeCare/Services/SummaryReport/CostBudgetsWorkSummary.cs (hunk 4 above)
Comment (AmrutaBorgaonkar) [reply BridgeCareApp/BridgeCare/Services/SummaryReport/CostBudgetsWorkSummary.cs]: Yes, that might create problem. Changing those to regular classes.

File: BridgeCareApp/BridgeCare/Services/SummaryReport/BridgeRateDeckAreaWorkSummary.cs (hunk 2 above)
Comment (AmrutaBorgaonkar) [reply BridgeCareApp/BridgeCare/Services/SummaryReport/BridgeRateDeckAreaWorkSummary.cs]: I used resources file for keeping all these const strings.

File: BridgeCareApp/BridgeCare/Services/SummaryReport/CostBudgetsWorkSummary.cs (hunk 4 above)
Comment (AmrutaBorgaonkar) [reply BridgeCareApp/BridgeCare/Services/SummaryReport/CostBudgetsWorkSummary.cs]: Updated.

**Comment by AmrutaBorgaonkar** [comment]
@ARA-Trans  Please have a look at it and let me know if it looks okay. Thank you.

File: BridgeCareApp/BridgeCare/Services/SummaryReport/BridgeWorkSummaryCommon.cs [hunk 5]
```diff
@@ -0,0 +1,145 @@
+﻿using BridgeCare.Models;
+using OfficeOpenXml;
+using System.Collections.Generic;
+
+namespace BridgeCare.Services
+{
+    public static class BridgeWorkSummaryCommon
```
[the remark this answers is not in the document]
Comment (AmrutaBorgaonkar) [reply BridgeCareApp/BridgeCare/Services/SummaryReport/BridgeWorkSummaryCommon.cs]: Done.

File: BridgeCareApp/BridgeCare/Services/SummaryReport/ExcelHelper.cs [hunk 6]
```diff
@@ -0,0 +1,83 @@
+﻿using System;
+using System.Drawing;
+using OfficeOpenXml;
+using OfficeOpenXml.Style;
+
+namespace BridgeCare.Services
+{
+    public static class ExcelHelper
```
[the remark this answers is not in the document]
Comment (AmrutaBorgaonkar) [reply BridgeCareApp/BridgeCare/Services/SummaryReport/ExcelHelper.cs]: Done.

**Comment by AmrutaBorgaonkar** [comment]
@ARA-Trans please have a look at changes and let me know if it looks good to complete this PR.
Thank you.

File: BridgeCareApp/BridgeCare/Models/SummaryReport/SimulationDataModel.cs [hunk 7]
```diff
@@ -7,5 +7,8 @@ public class SimulationDataModel
         public int SectionId { get; set; }
 
         public List<YearsData> YearsData { get; set; }
+
+        // Useful in some tab reports
```
Comment (ARA-Trans) [inline BridgeCareApp/BridgeCare/Models/SummaryReport/SimulationDataModel.cs]: Useful -> Used

Comment (AmrutaBorgaonkar) [reply BridgeCareApp/BridgeCare/Models/SummaryReport/SimulationDataModel.cs]: Done.

**Pull request merged**